x-pack/legacy/plugins/apm/public/new-platform/plugin.tsx

```diff
@@ -59,8 +63,17 @@ export interface ApmPluginStartDeps {
   data: DataPublicPluginStart;
 }
 
-const PluginsContext = createContext({} as ApmPluginStartDeps);
+export interface ConfigSchema {
+  apmIndexPatternTitle: string;
+  apmServiceMapEnabled: boolean;
+  apmUiEnabled: boolean;
+}
 
+// These are to be used until we switch over all our context handling to
+// kibana_react
+export const PluginsContext = createContext<
+  ApmPluginStartDeps & { apm: { config: ConfigSchema } }
+>({} as ApmPluginStartDeps & { apm: { config: ConfigSchema } });
 export function usePlugins() {
```

Comment:
Not necessary for this PR just an observation:

We normally have hooks in the [hooks folder](https://github.com/elastic/kibana/tree/cd109fa7c7593cf174b74474734ec8fabf20d0f5/x-pack/legacy/plugins/apm/public/hooks). Although for context hooks I think it's better to co-locate the hook with the context.

We don't currently do this, but I think it would be beneficial if the hooks signalled that it was consuming something from context - so a consumer will know they have to wrap the consuming component inside a context provider. In this case it could be called `usePluginsContext`.

Similarly `useLocation` should be renamed to `useLocationContext`.

Reply:
I think it makes more sense to leave these here as-is for now, since once we have this in place we can start moving to using kibana_react, and we'll be able to delete both the hook and the context.